src/sagas/wallet.js

```diff
@@ -456,11 +459,21 @@ export function* handleTx(action) {
   const tx = action.payload;
   const wallet = yield select((state) => state.wallet);
   const routerHistory = yield select((state) => state.routerHistory);
+  const receivedTxs = yield select((state) => state.receivedTxs);
 
   if (!wallet.isReady()) {
     return;
   }
 
+  if (receivedTxs.has(tx.tx_id)) {
```

Comment:
Should we have a special treatment if the first event was a `NEW_TX` and the second one an `UPDATE_TX`? Or this would not be necessary here?

Reply:
We can but I do not think we should right now.
The reason for this `receivedTxs` was because when we send a transaction we receive the same event multiple times for different reasons (once for every address in the tx, once when a block confirms the transaction, etc.)

These events would trigger multiple notifications for a each transaction and would request multiple history updates, which would make the UI of the wallet page "blink" while updating.
The main issue is that even though the event was being received as duplicate, our wallet-lib interprets it as 1 `new-tx` and multiple `update-tx` but this is not the case.

These issues will be resolved with the reliable integration eventually, but we currently should resolve the negative impact on the user experience (e.g. blinking pages, multiple notifications).